kore/src/Kore/Parser/Parser.y

```diff
@@ -0,0 +1,452 @@
+import Kore.Syntax.Sentence
+import Kore.Parser.Lexer
+import Kore.Parser.LexerWrapper
+import Numeric.Natural
+import qualified Prelude as Prelude
+import Prelude.Kore
+
+}
+
+%name aliasHeadStart AliasHead
+%name attributesStart Attributes
+%name definitionStart Definition
+%name elementVariableStart ElementVariable
+%name idStart Id
+%name moduleStart Module
+%name patternStart Pattern
+%name sentenceStart Sentence
+%name setVariableStart SetVariable
+%name sortStart Sort
+%name sortsParenStart SortsParen
+%name sortVariableStart SortVariable
+%name sortVariablesStart SortVariables
+%name stringLiteralStart StringLiteral
+%name symbolHeadStart SymbolHead
+
+%tokentype { Token }
+%monad { Alex }
+%lexer { alexMonadScan >>= } { Token _ TokenEOF }
+%error { happyError }
+
+%token
+    module               { Token _ TokenModule }
+    endmodule            { Token _ TokenEndModule }
+    import               { Token _ TokenImport }
+    sort                 { Token _ TokenSort }
+    symbol               { Token _ TokenSymbol }
+    where                { Token _ TokenWhere }
+    alias                { Token _ TokenAlias }
+    axiom                { Token _ TokenAxiom }
+    claim                { Token _ TokenClaim }
+    hookedSort           { Token _ TokenHookedSort }
+    hookedSymbol         { Token _ TokenHookedSymbol }
+    ':='                 { Token _ TokenColonEqual }
+    ':'                  { Token _ TokenColon }
+    '{'                  { Token _ TokenLeftBrace }
+    '}'                  { Token _ TokenRightBrace }
+    '['                  { Token _ TokenLeftBracket }
+    ']'                  { Token _ TokenRightBracket }
+    '('                  { Token _ TokenLeftParen }
+    ')'                  { Token _ TokenRightParen }
+    ','                  { Token _ TokenComma }
+    top                  { Token _ TokenTop }
+    bottom               { Token _ TokenBottom }
+    not                  { Token _ TokenNot }
+    and                  { Token _ TokenAnd }
+    or                   { Token _ TokenOr }
+    implies              { Token _ TokenImplies }
+    iff                  { Token _ TokenIff }
+    exists               { Token _ TokenExists }
+    forall               { Token _ TokenForall }
+    mu                   { Token _ TokenMu }
+    nu                   { Token _ TokenNu }
+    ceil                 { Token _ TokenCeil }
+    floor                { Token _ TokenFloor }
+    equals               { Token _ TokenEquals }
+    in                   { Token _ TokenIn }
+    next                 { Token _ TokenNext }
+    rewrites             { Token _ TokenRewrites }
+    dv                   { Token _ TokenDomainValue }
+    leftAssoc            { Token _ TokenLeftAssoc }
+    rightAssoc           { Token _ TokenRightAssoc }
+    ident                { Token _ (TokenIdent _) }
+    setIdent             { Token _ (TokenSetIdent _) }
+    string               { Token _ (TokenString _) }
```

Comment:
```suggestion
    ident                { Token _ (TokenIdent $$) }
    identLoc             { Token _ (TokenIdent _) } -- variation which keeps AlexPn
    setIdent             { Token _ (TokenSetIdent $$) }
    string               { Token _ (TokenString $$) }
```
Now using `$n` in an action to refer to these tokens refers to the text contents. I think this allows you to delete `getTokenBody` entirely.